strapi/strapi
# Check the code versioning before starting the upgrade process

### What does it do?

It checks if the code is under a git versioning and if it is clean before continuing with the update process

### Why is it needed?

To make the upgrade safer

### How to test it?

- Run the upgrade in a process with a dirty git tree and see that there is a warning and asks you if you want to continue
- Run the upgrade in a project with no Git VCS and see there is a warning and asks you if you want to continue
- Run the upgrade with --force option and it should continue no matter the 2 previous scenarios

### Related issue(s)/PR(s)
